### PR TITLE
test: VM id updated after snapshotting and storage pool copy

### PR DESCRIPTION
**Pull request by gabrielmougard**
Depends on https://github.com/lxc/lxd/pull/11385

### CODE REVIEW

**Comment by stgraber** [comment]
@tomponline any thoughts on what testsuite this should go into?
I'd rather not have super small tests like this one spin up dedicated MAAS instances for such a short test.

**Comment by tomponline** [comment]
@stgraber it could go in https://github.com/lxc/lxc-ci/blob/master/bin/test-lxd-storage-vm as that does snapshots and restores of VMs already.

**Comment by stgraber** [comment]
Sounds good.
@gabrielmougard can you make that change?

**Comment by gabrielmougard** [comment]
@stgraber sure

**Comment by tomponline** [comment]
@gabrielmougard I've updated the PR's description to cross-link to the LXD PR that this PR depends on.

File: bin/test-lxd-storage-vm [hunk 1]
```diff
@@ -420,6 +420,56 @@ do
...
+        lxc restore v1 snap1
+        newUUID=$(lxc config show v1 --expanded | grep "volatile.uuid:" | awk '{print $2}')
+
+        if [ "${initialUUID}" = "${newUUID}" ]; then
+                echo "==> UUID of the VM instance should be different after restoring its snapshot"
+                false
+        fi
+
+        echo "==> Checking VM UUID while snapshotting (stateful mode)"
+        lxc launch images:busybox/1.34.1 v2 --vm -s "${poolName}" --ephemeral
```
Comment (tomponline) [inline bin/test-lxd-storage-vm:440]: Please use `images:ubuntu/jammy` for VM tests.

You only need to use busybox for jenkins test suite tests and/or container stateful tests.
But this test is only for VMs so we can use a normal Ubuntu image.

Comment (tomponline) [inline bin/test-lxd-storage-vm:440]: Any reason for the `--ephemeral` use here?

**Comment by tomponline** [comment]
@gabrielmougard did you run this script locally (or on a maas machine) using the instructions here https://discuss.linuxcontainers.org/t/how-to-create-lxd-binaries-from-the-source-code-and-side-load-them-in-an-existing-snap-installation/16551 to check this works with your branch?

**Comment by gabrielmougard** [comment]
> @gabrielmougard did you run this script locally (or on a maas machine) using the instructions here https://discuss.linuxcontainers.org/t/how-to-create-lxd-binaries-from-the-source-code-and-side-load-them-in-an-existing-snap-installation/16551 to check this works with your branch?

I messed up lxd with snap I think..

**Comment by gabrielmougard** [comment]
@tomponline I'm still on it.. had to face some issues on my laptop regarding snap

File: bin/test-lxd-storage-vm [hunk 2]
```diff
@@ -420,6 +420,57 @@ do
         lxc image delete vmbig
         lxc profile delete vmsmall
 
+        echo "==> Checking VM UUID while snapshotting (non-stateful mode)"
+        lxc init images:ubuntu/20.04/cloud v1 --vm -s "${poolName}"
```
Comment (tomponline) [inline bin/test-lxd-storage-vm:424]: Should use 22.04 now.

File: bin/test-lxd-storage-vm [hunk 3]
```diff
@@ -420,6 +420,57 @@ do
...
+        lxc start v1
+        waitVMAgent v1
+        lxc info v1
+
+        initialUUID=$(lxc config show v1 --expanded | grep "volatile.uuid:" | awk '{print $2}')
```
Comment (tomponline) [inline bin/test-lxd-storage-vm:429]: `lxc config get v1 volatile.uuid` ...again ;)

File: bin/test-lxd-storage-vm [hunk 4]
```diff
@@ -420,6 +420,57 @@ do
...
+
+        if [ "${initialUUID}" = "${newUUID}" ]; then
+                echo "==> UUID of the VM instance should be different after restoring its snapshot"
+                false
+        fi
+
+        lxc delete -f v1
+
+        echo "==> Checking VM UUID while snapshotting (stateful mode)"
+        lxc init images:ubuntu/jammy v2 --vm -s "${poolName}"
```
Comment (tomponline) [inline bin/test-lxd-storage-vm:442]: Please use release numbers or names consistently. And dont mix versions.

File: bin/test-lxd-storage-vm (hunk 3 above)
Comment (gabrielmougard) [reply bin/test-lxd-storage-vm:429]: ah yeah I missed this one

File: bin/test-lxd-storage-vm [hunk 5]
```diff
@@ -420,6 +420,57 @@ do
...
+        lxc config set v2 migration.stateful=true
+        lxc config device set v2 root size.state=2GiB
```
Comment (tomponline) [inline bin/test-lxd-storage-vm:444]: Why not 1GiB as that is the default memory size?

File: bin/test-lxd-storage-vm [hunk 6]
```diff
@@ -420,6 +420,57 @@ do
...
+        echo "==> Checking VM UUID while snapshotting (non-stateful mode)"
+        lxc init images:ubuntu/20.04/cloud v1 --vm -s "${poolName}"
+        lxc start v1
+        waitVMAgent v1
+        lxc info v1
+
+        initialUUID=$(lxc config show v1 --expanded | grep "volatile.uuid:" | awk '{print $2}')
+        lxc snapshot v1 snap1
+        lxc restore v1 snap1
+        newUUID=$(lxc config show v1 --expanded | grep "volatile.uuid:" | awk '{print $2}')
```
Comment (tomponline) [inline bin/test-lxd-storage-vm:432]: lxc config get v1 volatile.uuid

**Review by tomponline — CHANGES_REQUESTED** [review]
A few nits, but looks good.

**Comment by tomponline** [comment]
Thanks!

**Pull request merged**